dynamic_rest/filters.py

```diff
@@ -691,22 +686,66 @@ def remove_invalid_fields(self, queryset, fields, view):
             stripped_term = term.lstrip('-')
             # add back the '-' add the end if necessary
             reverse_sort_term = '' if len(stripped_term) is len(term) else '-'
-            if stripped_term in valid_fields_map:
-                name = reverse_sort_term + valid_fields_map[stripped_term]
-                valid_orderings.append(name)
+            ordering = self.ordering_for(stripped_term, view)
+
+            if ordering:
+                valid_orderings.append(reverse_sort_term + ordering)
             else:
                 invalid_orderings.append(term)
 
         return valid_orderings, invalid_orderings
 
-    def get_valid_fields(self, queryset, view, context={}):
-        """Return valid fields for ordering.
+    def ordering_for(self, term, view):
+        """
+        Return ordering (model field chain) for term (serializer field chain)
+        or None if invalid
 
-        Overwrites DRF's get_valid_fields method so that valid_fields returns
-        serializer fields, not model fields.
+        Raise ImproperlyConfigured if serializer_class not set on view
         """
+        if not self._is_allowed_term(term, view):
+            return None
+
+        serializer = self._get_serializer_class(view)()
+        serializer_chain = term.split('.')
+
+        if len(serializer_chain) == 1:
```

Comment:
I believe this `if ... else` block is unnecessary, and what's currently in the `else` block should work in all cases (if `serializer_chain` is size of 1, the for loop is simply skipped, but the rest should do the right thing).

Reply:
I didn't actually read through this originally since I based this off of @Incanus3's commit, but yeah, great point. The `if ... else` block has been removed and I've kept the `else` section.